SignalServiceKit/src/Loki/Protocol/Session Management/LokiSessionResetImplementation.swift

```diff
@@ -45,19 +46,28 @@ public class LokiSessionResetImplementation : NSObject, SessionResetProtocol {
             return
         }
         guard !recipientID.isEmpty else { return }
+        let masterHexEncodedPublicKey = storage.getMasterHexEncodedPublicKey(for: recipientID, in: transaction) ?? recipientID
         guard let thread = TSContactThread.getWithContactId(recipientID, transaction: transaction) else {
             Logger.debug("[Loki] A new session was adopted but the thread couldn't be found for: \(recipientID).")
             return
         }
         // If the current user initiated the reset then send back an empty message to acknowledge the completion of the session reset
-        if thread.sessionResetStatus == .initiated {
+        if let masterThread = TSContactThread.getWithContactId(masterHexEncodedPublicKey, transaction: transaction), masterThread.sessionResetStatus == .initiated {
             let emptyMessage = EphemeralMessage(thread: thread)
-            SSKEnvironment.shared.messageSender.sendPromise(message: emptyMessage).retainUntilComplete()
+            SSKEnvironment.shared.messageSender.sendPromise(message: emptyMessage).done2{ _ in
```

Comment:
This'll work but it'd be better to use the message sender job queue to ensure that things still proceed even if the user closes the app before the message is sent